lib/elixir_analyzer/exercise_test/assert_call/compiler.ex

```diff
@@ -195,4 +230,47 @@ defmodule ElixirAnalyzer.ExerciseTest.AssertCall.Compiler do
   """
   def in_function?(name, {_module_path, name}), do: true
   def in_function?(_, _), do: false
+
+  defp track_imports({:import, _, [module_path | _]}),
+    do: get_import_paths(module_path)
+
+  defp track_imports(_), do: []
+
+  defp get_import_paths({:__aliases__, _, path}),
+    do: [{path, path}]
+
+  defp get_import_paths({{:., _, [root, :{}]}, _, branches}) do
+    [{root_path, _}] = get_import_paths(root)
+
+    for branch <- branches,
+        {path, _} <- get_import_paths(branch) do
+      {root_path ++ path, root_path ++ path}
+    end
+  end
+
+  defp get_import_paths(path) when is_atom(path),
+    do: [{[path], [path]}]
+
+  defp track_aliases({:alias, _, [module_path]}),
+    do: get_alias_paths(module_path)
+
+  defp track_aliases({:alias, _, [module_path, [as: {:__aliases__, _, [alias]}]]}),
+    do: get_alias_paths(module_path) |> Enum.map(fn {_, path} -> {[alias], path} end)
+
+  defp track_aliases(_), do: []
+
+  defp get_alias_paths({:__aliases__, _, path}),
+    do: [{[List.last(path)], path}]
+
+  defp get_alias_paths({{:., _, [root, :{}]}, _, branches}) do
+    [{_, root_path}] = get_alias_paths(root)
+
+    for branch <- branches,
+        {last, full_path} <- get_alias_paths(branch) do
+      {last, root_path ++ full_path}
+    end
+  end
+
+  defp get_alias_paths(path) when is_atom(path),
+    do: [{[path], [path]}]
```

Comment:
Following the suggestion above to extract the`Enum.reduce` to a private function to be reused by `track_import/2` and `track_alias/2`.  I would prefer this style  because it means to understand `annotate` we don't also have to decipher the `Enum.reduce` expression now found in `track_modules/2`.

Also, while I may not be 100% accurate in my own coding, if one function clause has required the multi-line function block once, they should all be multiline blocks for consistency.

```suggestion
  # track_imports
  defp track_imports(acc, {:import, _, [module_path | _]}) do
    paths = get_import_paths(module_path)
    track_modules(acc, paths)
  end

  defp track_imports(acc, _) do
    acc
  end

  # get_import_paths
  defp get_import_paths({:__aliases__, _, path}) do
    [{path, path}]
  end

  defp get_import_paths({{:., _, [root, :{}]}, _, branches}) do
    [{root_path, _}] = get_import_paths(root)

    for branch <- branches,
        {path, _} <- get_import_paths(branch) do
      {root_path ++ path, root_path ++ path}
    end
  end

  defp get_import_paths(path) when is_atom(path) do
    [{[path], [path]}]
  end

  # track_aliases
  defp track_aliases(acc, {:alias, _, [module_path]}) do
    paths = get_alias_paths(module_path)
    track_modules(acc, paths)
  end

  defp track_aliases(acc, {:alias, _, [module_path, [as: {:__aliases__, _, [alias]}]]}) do
    paths = get_alias_paths(module_path) |> Enum.map(fn {_, path} -> {[alias], path} end)
    track_modules(acc, paths)
  end

  defp track_aliases(acc, _) do
    acc
  end

  # get_alias_paths
  defp get_alias_paths({:__aliases__, _, path}) do
    [{[List.last(path)], path}]
  end

  defp get_alias_paths({{:., _, [root, :{}]}, _, branches}) do
    [{_, root_path}] = get_alias_paths(root)

    for branch <- branches,
        {last, full_path} <- get_alias_paths(branch) do
      {last, root_path ++ full_path}
    end
  end

  defp get_alias_paths(path) when is_atom(path) do
    [{[path], [path]}]
  end

  # track modules
  defp track_modules(acc, module_paths) do
    Enum.reduce(module_paths, acc, fn {alias, full_path}, acc ->
      if acc.in_function_def,
        do: %{acc | in_function_modules: Map.put(acc.in_function_modules, alias, full_path)},
        else: %{acc | modules_in_scope: Map.put(acc.modules_in_scope, alias, full_path)}
    end)
  end
```